### PR TITLE
Users can clear their mailboxes (Updated)

### PR DESCRIPTION
**Pull request by sazanof**
**Hello!** 
Now when user clear mailbox and if this mailbox not equals Trash, messages move to trash firstly

Referenced https://github.com/nextcloud/mail/pull/6286 , https://github.com/nextcloud/mail/pull/6768#issuecomment-1167040130

Fixes https://github.com/nextcloud/mail/issues/5858

### CODE REVIEW

File: lib/Db/MessageMapper.php [hunk 1]
```diff
@@ -1220,4 +1220,16 @@ public function resetInReplyTo(): int {
 			);
 		return $update->execute();
 	}
+
+	public function changeMailbox(int $currentId, int $destinationId): int {
+		$qb = $this->db->getQueryBuilder();
+
+		$update = $qb->update($this->tableName)
```
Comment (ChristophWurst) [inline lib/Db/MessageMapper.php:1227]: this won't work. each IMAP message has a `UID` assigned, but that UID is only unique within the mailbox. If a message is moved from one mailbox to another, a new UID is assigned.

If you only update the mailbox_id, the message UIDs will point to other messages.

It is not trivial to figure out the new UIDs. Therefore I would suggest that we move the messages on IMAP and delete them from the source messages cache. Synchronization will re-add the messages in the trash bin with their new UIDs.

File: lib/Service/MailManager.php [hunk 2]
```diff
@@ -585,6 +585,40 @@ public function deleteMailbox(Account $account,
...
+	 *
+	 * @throws DoesNotExistException
+	 * @throws Horde_Imap_Client_Exception
+	 * @throws Horde_Imap_Client_Exception_NoSupportExtension
+	 * @throws ServiceException
+	 */
+	public function clearMailbox(Account $account,
+								  Mailbox $mailbox): void {
+		$client = $this->imapClientFactory->getClient($account);
+		$trashMailboxId = $account->getMailAccount()->getTrashMailboxId();
```
Comment (ChristophWurst) [inline lib/Service/MailManager.php:602]: ``$trashMailboxId`` can be `null`

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: lib/Service/MailManager.php (hunk 2 above)
Comment (sazanof) [reply lib/Service/MailManager.php:602]: So if null, then just delete permanently, right?

File: lib/Db/MessageMapper.php (hunk 1 above)
Comment (sazanof) [reply lib/Db/MessageMapper.php:1227]: OK, just do after that `$this->dbMessageMapper->deleteAll($mailbox);`

File: lib/Service/MailManager.php (hunk 2 above)
Comment (ChristophWurst) [reply lib/Service/MailManager.php:602]: Yes :)

Comment (sazanof) [reply lib/Service/MailManager.php:602]: This method is deleted

File: src/components/NavigationMailbox.vue [hunk 3]
```diff
@@ -507,6 +519,29 @@ export default {
 				this.clearCache = false
 			}
 		},
+		clearMailbox() {
+			const id = this.mailbox.databaseId
+			OC.dialogs.confirmDestructive(
+				t('mail', 'All messages in folder will be deleted.'),
```
Comment (ChristophWurst) [inline src/components/NavigationMailbox.vue:525]:
```suggestion
				t('mail', 'All messages in mailbox will be deleted.'),
```

**Comment by sazanof** [comment]
Good afternoon! =) I made changes. Do you have any other requests? 
If you have time, can you double-check the PR? Thanks!

**Comment by sazanof** [comment]
@ChristophWurst hello! 
I fixed the conflicts! =) can you see when there will be time? Thank you and have a nice day!

**Review by ChristophWurst — APPROVED** [review]
Tested, works

**Comment by ChristophWurst** [comment]
```
/home/runner/work/mail/mail/src/service/MailboxService.js
Error:   58:34  error  Missing space before function parentheses  space-before-function-paren
```

**Pull request merged**